### PR TITLE
OSPF validation helpers

### PR DESCRIPTION
**Pull request by duplico**
## Summary

Implements three OSPF validation helpers for functional testing as specified in #144. These helpers verify OSPF configuration state via VyOS operational commands.

**New Validation Helpers:**
1. `check_ospf_enabled(ssh)` - Verify OSPF process is running
2. `check_ospf_router_id(ssh, expected_id)` - Verify OSPF router ID matches
3. `check_ospf_interface(ssh, interface, area)` - Verify interface OSPF configuration

## Changes

**Modified Files:**
- `tests/validation_helpers.py` - Added three OSPF validation helpers
- `tests/test_validation_helpers.py` - Added 17 unit tests for OSPF helpers

**VyOS Commands Used:**
- `show ip ospf` - Check OSPF process status and router ID
- `show configuration commands | grep ospf` - Validate interface OSPF configuration

## Test Coverage

All OSPF helpers include comprehensive unit tests with mocked SSH responses:
- Success cases with expected values
- Failure cases (OSPF not running, mismatch, not found)
- SSH command failure handling
- Edge cases (empty output, unexpected formats, multiple interfaces)

**Total Test Coverage:** 47 tests (30 from PR #159 base + 17 new OSPF tests), all passing.

## Test Plan

- [x] `just check` passes (ruff, mypy, pytest)
- [x] All 47 unit tests pass with mocked SSH
- [x] OSPF helpers parse VyOS output correctly
- [x] Error handling works for SSH failures
- [x] Docstrings document VyOS output formats
- [x] Follows patterns from PR #159

## Dependencies

- **Addresses:** #144
- **Depends on:** #159 (core validation infrastructure)
- **Part of epic:** #86 (Functional testing framework)
- **Blocks:** #102 (ospf.env validation test)

## Notes

These helpers verify OSPF **configuration**, not neighbor adjacency (which requires multi-VM testing). The helpers work with single-VM QEMU tests where OSPF is configured but neighbors may not be present.

Generated with Claude Code (Sonnet 4.5)

### CODE REVIEW

File: tests/validation_helpers.py [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+- Takes an ssh_connection callable (from conftest.py)
+- Executes VyOS operational mode commands
+- Returns a ValidationResult with pass/fail status and diagnostic info
+
+VyOS Command Output Formats:
+- show interfaces: Contains "link/ether", "inet" lines with IP addresses
+- show configuration | grep host-name: Returns "host-name 'hostname'"
+- show configuration | grep public-keys: Returns public-keys stanzas if configured
+- show ip ospf: Shows OSPF instance status including router ID
+- show ip ospf interface: Shows interface-level OSPF configuration
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/validation_helpers.py]: Fixed in 86fad8f. Removed the unused 'show ip ospf interface' command from the module-level docstring to keep it accurate with the actual commands used by the helper functions.

(AI-generated via Claude Code w/ Claude Haiku 4.5)
